cmd/earthly/main.go

```diff
@@ -2944,11 +2944,22 @@ func (app *earthlyApp) actionBuildImp(c *cli.Context, flagArgs, nonFlagArgs []st
 	buildContextProvider.AddDirs(defaultLocalDirs)
 	attachables := []session.Attachable{
 		llbutil.NewSecretProvider(cc, secretsMap),
-		authprovider.NewDockerAuthProvider(os.Stderr),
 		buildContextProvider,
 		localhostProvider,
 	}
 
+	switch app.containerFrontend.Config().Setting {
+	case containerutil.FrontendDocker, containerutil.FrontendDockerShell:
+		attachables = append(attachables, authprovider.NewDockerAuthProvider(os.Stderr))
+
+	case containerutil.FrontendPodman, containerutil.FrontendPodmanShell:
+		attachables = append(attachables, authprovider.NewPodmanAuthProvider(os.Stderr))
+
+	default:
```

Comment:
Nit: Do we need the `default` case? Might be more concise as if/else:
```golang
if containerutil.FrontendPodman || containerutil.FronendPodmanShell {
   // .. 
} else {
   // docker
}
```

Reply:
We may get more frontends in the future (see things like nerdctl), and it really explicitly enumerates why each part is there. I prefer this over the `if` for explicitness in the _why_ each thing is there.